mmocr/models/common/backbones/clip_resnet.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+import torch.nn as nn
+from mmdet.models.backbones import ResNet
+from mmdet.models.backbones.resnet import Bottleneck
+
+from mmocr.registry import MODELS
+
+
+class CLIPBottleneck(Bottleneck):
+    """Bottleneck for CLIPResNet.
+
+    It is a variant Bottleneck used in the variant ResNet of CLIP. After the
+    second convolution layer, there is an additional average pooling layer with
+    kernel_size 2 and stride 2, which is added in the plug-in manner when the
+    input stride > 1. The stride of each convolution layer is always set to 1.
+    """
+
+    def __init__(self, **kwargs):
+        stride = kwargs.get('stride', 1)
+        kwargs['stride'] = 1
+        plugins = kwargs.get('plugins', None)
+        if stride > 1:
+            if plugins is None:
+                plugins = []
+
+            plugins.insert(
+                0,
+                dict(
+                    cfg=dict(type='mmocr.AvgPool2d', kernel_size=2),
+                    position='after_conv2'))
+            kwargs['plugins'] = plugins
+        super().__init__(**kwargs)
+
+
+@MODELS.register_module()
+class CLIPResNet(ResNet):
+    """Implement the variant ResNet used in `oCLIP.
+
+    <https://github.com/bytedance/oclip>`_
+
+    It is also the official structure in `CLIP
+    <https://github.com/openai/CLIP>`_.
+
+    Compared with ResNetV1d structure, CLIPResNet replaces the
+    max pooling layer with an average pooling layer at the end
+    of the input stem.
+
+    In the Bottleneck of CLIPResNet, after the second convolution
+    layer, there is an additional average pooling layer with
+    kernel_size 2 and stride 2, which is added in the plug-in
+    manner when the input stride > 1.
+    The stride of each convolution layer is always set to 1.
+    """
```

Comment:
```python
Args:
    depth (int): xxx
```
referring to https://github.com/open-mmlab/mmdetection/blob/9c1b26726eebe4a196d213249dc22e8017761fab/mmdet/models/backbones/resnet.py#L309

Reply:
done